### PR TITLE
Cleanup of Dialogs

### LINKED ISSUE #50: Get rid of `AlertFragment` in favor of composable `AlertFragmentDialog`

### PR DESCRIPTION
**Pull request by ArnyminerZ**

### CODE REVIEW

**Review by sunkup — CHANGES_REQUESTED** [review]
Does not work :(

Now every time I enter the EditSubscriptionActivity, it tells me there are unsaved changes.

**Comment by ArnyminerZ** [comment]
My bad, I had a stupid typo

File: app/src/main/java/at/bitfire/icsdroid/ui/dialog/AlertDialogBox.kt
Comment (sunkup) [inline app/src/main/java/at/bitfire/icsdroid/ui/dialog/AlertDialogBox.kt:1]: I don't think using `AlertDialogBox` to wrap `GenericAlertDialog` while adding only the Text() composable is worth it. I think we can safely use `GenericAlertDialog` everywhere and delete this file.

File: app/src/main/java/at/bitfire/icsdroid/ui/dialog/GenericAlertDialog.kt [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    content: (@Composable AlertDialogContext.() -> Unit)? = null,
+    onDismissRequest: () -> Unit
+) {
+    val context = object : AlertDialogContext {
+        override fun dismiss() = onDismissRequest()
+    }
+
+    AlertDialog(
+        onDismissRequest = onDismissRequest,
+        // FIXME : I don't love this syntax, but I'm not sure if there's a cleaner way...
```
Comment (sunkup) [inline app/src/main/java/at/bitfire/icsdroid/ui/dialog/GenericAlertDialog.kt]: I don't think it's too bad :) Nearly looks like handlebar template engine :+1: 

Would probably make them into one liners where possible though ;)

File: app/src/main/java/at/bitfire/icsdroid/ui/dialog/SyncIntervalDialog.kt [hunk 2]
```diff
@@ -1,31 +1,68 @@
...
+import androidx.compose.ui.Modifier
 import androidx.compose.ui.platform.LocalContext
 import androidx.compose.ui.res.stringArrayResource
+import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.tooling.preview.Preview
 import at.bitfire.icsdroid.R
 import com.google.android.material.dialog.MaterialAlertDialogBuilder
 
 @Composable
+@OptIn(ExperimentalMaterialApi::class)
```
Comment (sunkup) [inline app/src/main/java/at/bitfire/icsdroid/ui/dialog/SyncIntervalDialog.kt:30]: Wow. Why are list items still expermental? :D

File: app/src/main/java/at/bitfire/icsdroid/ui/dialog/GenericAlertDialog.kt [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+import androidx.compose.material.AlertDialog
+import androidx.compose.material.Text
+import androidx.compose.material.TextButton
+import androidx.compose.runtime.Composable
+
+/**
+ * Provided by [GenericAlertDialog], gives access to some interactions from inside the composables
+ * and callbacks.
+ */
+interface AlertDialogContext {
```
Comment (sunkup) [inline app/src/main/java/at/bitfire/icsdroid/ui/dialog/GenericAlertDialog.kt]: Really cool, that this is possible and good to know. But I feel the complexity it adds in comparison do the benefit it brings is too small. I would keep this for more complicated compose setups and stick with the simple tools for now.

**Review by sunkup — CHANGES_REQUESTED** [review]
See comments :)

**Review by sunkup — APPROVED** [review]
Nice ~

**Pull request merged**